src/main/java/com/snowflake/kafka/connector/internal/streaming/SnowflakeSinkServiceV2.java

```diff
@@ -256,15 +261,36 @@ public void callAllGetOffset() {
 
   @Override
   public void closeAll() {
-    partitionsToChannel.forEach(
-        (partitionChannelKey, topicPartitionChannel) -> {
-          LOGGER.info("Closing partition channel:{}", partitionChannelKey);
-          topicPartitionChannel.closeChannel();
-        });
+    LOGGER.info(
+        "Closing All partitions:{}", Arrays.toString(partitionsToChannel.keySet().toArray()));
+    closeStreamingIngestChannels(partitionsToChannel.values());
     partitionsToChannel.clear();
     closeStreamingClient();
   }
 
+  /** Close Streaming Channels associated to each of the TopicPartition Channel Objects. */
+  private void closeStreamingIngestChannels(
+      Collection<TopicPartitionChannel> topicPartitionChannels) {
+    List<CompletableFuture> partitionCloseFutures = new ArrayList<>();
+    topicPartitionChannels.forEach(
+        topicPartitionChannel -> partitionCloseFutures.add(topicPartitionChannel.closeChannel()));
+    try {
+      // waits for all channels to close and eventually does a join on Future result. (Returned by
+      // allOf)
+      CompletableFuture.allOf(
+              partitionCloseFutures.toArray(new CompletableFuture[partitionCloseFutures.size()]))
+          .join();
+    } catch (CancellationException cancellationException) {
+      LOGGER.warn(
+          String.format(
+              "Close cancelled for partitions:%s",
+              Arrays.toString(topicPartitionChannels.toArray())),
+          cancellationException);
+    } catch (CompletionException completionException) {
+      LOGGER.warn("Close failed with exception", completionException);
```

Comment:
I see CompletionException because one of the channels didnt complete successfully even after 17 seconds. 
https://github.com/snowflakedb/snowflake-kafka-connector/pull/440#issuecomment-1146728290